extensions/cache/runtime/src/main/java/io/quarkus/cache/runtime/AbstractCache.java

```diff
@@ -0,0 +1,54 @@
+package io.quarkus.cache.runtime;
+
+import java.util.Objects;
+
+import io.quarkus.cache.Cache;
+import io.quarkus.cache.runtime.caffeine.CaffeineCache;
+
+public abstract class AbstractCache implements Cache {
```

Comment:
is that a user public contract?

Reply:
if not let's move things under .impl

Reply:
Fixed.

Reply:
Moving many classes from `.runtime` to `.impl` actually made the review harder so I'll fix this specific point later in a separate PR.